components/ui/autocomplete/build.gradle

```diff
@@ -0,0 +1,50 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+
+android {
+    compileSdkVersion rootProject.ext.compileSdkVersion
+
+    defaultConfig {
+        minSdkVersion rootProject.ext.minSdkVersion
+        targetSdkVersion rootProject.ext.targetSdkVersion
+        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+    }
+
+    buildTypes {
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+
+    testOptions {
+        unitTests {
+            includeAndroidResources = true
+        }
+    }
+}
+
+dependencies {
+    compile "com.android.support:appcompat-v7:27.1.0"
```

Comment:
I wonder if we should make this 'compileOnly'. This component might depend on appcompat (does it?), but we do not want to package it and leave it to the consumer to define which version of appcompat to add.

This should at least allow the app to use any 27.* in this case. Newer major versions (28.*) might make problems until we compile against them too. We probably want to centralize the android-support version in the root configuration so that we can make the promise that all components x.y work with support libraries q.z.

Reply:
👍  will update.

Reply:
I filed #37 for centralizing the configuration of the version. :)

Reply:
I have troubles moving this to compileOnly. The tests won't run (find the classes) even if I add the dependency as testImplementation as well.

Reply:
Yeah, I had this problem in one of the other modules too. For actually *running* we need to compile it into the test artifact (testImplementation). The duplication is a bit annoying... but I think makes sense.

Reply:
Right, but for me adding testImplementation as well doesn't work. So, not clear to me what's wrong. Still investigating...

Reply:
@pocmo OK, I've addressed all other issues. This one, I am stuck on right now. I can't get the tests to run with appcompat on compileOnly.

Reply:
So, at least I've found the cause now. When set to compileOnly, the R file for appcompat isn't being generated.  So ./build/generated/source/r/debug/android/support/v7/appcompat/R.java doesn't exist which causes the class loading error I was seeing. Need to look at this with fresh eyes tmw morning. If you have any ideas, please let me know :)

Reply:
Oh. Yeah, I guess this makes sense: Those are plain Java unit tests and therefore a test dependency will probably only add Java code to the classpath and ignore all other Android resources coming with the dependency.

This seems to be caused by the new code that tries to read the accent color instead of using one defined in XML like before, right? I see multiple options here:

* (1) Do not use the accent color. Instead define a custom attribute that accepts a color ([Docs](https://developer.android.com/training/custom-views/create-view.html#customattr)). Maybe this is even nicer because you might want to set this to something that is not the accent color of the app.

* (2) Our minSdkVersion is 21 (= Lollipop). The code trying to read this attribute from appcompat is not needed. You can just always read android.R.attr.colorAccent directly. You can even make this more explicit by adding an annotation to the class: @TargetApi(Build.VERSION_CODES.LOLLIPOP)

Reply:
I just tried option 2 and this resulted in a large batch of more "Unresolved reference" errors. Argh.

In the worst case we will have to go back to using "implementation" for all dependencies again. A consumer of the library has still the option to exclude/override the transitive dependency if they wish to use something else. For example [here in Focus we exclude support-annotations from the screengrab dependency](https://github.com/mozilla-mobile/focus-android/blob/master/app/build.gradle#L361-L363) because we already use a different version in our list of dependencies. I still think compileOnly would be nicer though. I'll do some research to see how other UI libraries solve this problem.

Reply:
OK, thanks for taking a look. I agree that compileOnly would be best. I'll give option 1 a try. I think it would be best to let devs freely define the color. We can fallback to our current color if none was defined.

Reply:
I tried using 'implementation' and forcing a conflict with a test app using the component.

If the app defines a higher version then gradle uses just that (library 26, app: 27):
> com.android.support:appcompat-v7:26.1.0 -> 27.1.0 (*)

If the library uses a higher version than gradle throws an error and forces you to resolve the conflict yourself:
> Android dependency 'com.android.support:appcompat-v7' has different version for the compile (26.1.0) and runtime (27.1.0) classpath. You should manually set the same version via DependencyResolution

In the app I can then decide to either upgrade to at least the version the library requires or exclude the transitive dependencies and continue to use my version (with potentially incompatibilities):
```groovy
 implementation(project(':ui-autocomplete'), {
        exclude group: 'com.android.support', module: 'appcompat-v7'
        exclude group: 'com.android.support', module: 'support-compat'
})
```

Maybe using "implementation" isn't that bad after all...?

Reply:
@pocmo OK, I've updated the commit to use `implementation` and also make the autocomplete color customizable. I think this is ready to go now, but I will leave so you can have another look before we merge this ;).